src/hooks/useRequireOnboarding.ts

```diff
@@ -0,0 +1,29 @@
+import { useUser } from "@auth0/nextjs-auth0/client";
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+
+// 500 and 404 pages are included to avoid an infinite redirect loop
+// should the onboarding page break
+const skipFor = ["/onboarding", "/500", "/404"];
+
+/**
+ * Redirects the user to onboarding after login
+ * when they have not previously completed onboarding
+ */
+export function useRequireOnboarding() {
```

Comment:
is this something we could do in middleware if we can figure out the env situation?

Reply:
Yeah, that is what i'm hoping 👍🏻